src/families/cosmos/mock.js

```diff
@@ -0,0 +1,170 @@
+ * @memberof cosmos/mock
+ * @param {Account} account
+ * @param {Prando} rng
+ */
+function genDelegationOperation(account: Account, rng: Prando): Account {
+  const { spendableBalance, operations: ops } = account;
+
+  if (spendableBalance.isZero()) return account;
+
+  const address = genAddress(account.currency, rng);
+
+  const value = spendableBalance;
+
+  /** select position on the operation stack where we will insert the new delegation */
+  const opIndex = rng.next(0, 10);
+
+  const lastOp = ops[opIndex];
+  const date = new Date(
+    (lastOp ? lastOp.date.valueOf() : Date.now()) -
+      rng.nextInt(0, 100000000 * rng.next() * rng.next())
+  );
+
+  /** select between 3 to 5 validators and split the amount evenly */
+  const delegatedValidators = Array(rng.nextInt(3, 5))
+    .fill(null)
+    .map(() => rng.nextArrayItem(validators))
+    .filter(
+      (validator, index, arr) =>
+        arr.findIndex(
+          v => v.validatorAddress === validator.validatorAddress
+        ) === index
+    )
+    .map(({ validatorAddress }, i, arr) => ({
+      address: validatorAddress,
+      amount: BigNumber(
+        Math.round(value.toNumber() * rng.next(0.1, 1 / arr.length))
+      )
+    }));
+
+  const extra: CosmosDelegateTxInfo = { validators: delegatedValidators };
+
+  /** format delegations and randomize rewards and status */
+  const delegations: CosmosDelegation[] = delegatedValidators.map(
+    ({ address, amount }) => ({
+      validatorAddress: address,
+      amount,
+      pendingRewards: rng.nextBoolean()
+        ? BigNumber(Math.round(amount.toNumber() * 0.01))
+        : BigNumber(0),
+      status: rng.next() > 0.33 ? "bonded" : "unbonded"
+    })
+  );
+
+  /** format cosmosResources given the new delegations */
+  const cosmosResources: CosmosResources = {
+    delegations,
+    delegatedBalance: delegations.reduce(
+      (sum, { amount }) => sum.plus(amount),
+      BigNumber(0)
+    ),
+    pendingRewardsBalance: delegations.reduce(
+      (sum, { pendingRewards }) => sum.plus(pendingRewards),
+      BigNumber(0)
+    ),
+    unboundingBalance: account.cosmosResources
+      ? account.cosmosResources.unboundingBalance
+      : BigNumber(0),
+    withdrawAddress: account.id
+  };
+
+  const fee = BigNumber(
+    Math.round(cosmosResources.delegatedBalance.toNumber() * 0.001)
+  );
+
+  const hash = genHex(64, rng);
+  /** generate given operation */
+  const op: $Exact<CosmosOperation> = {
+    id: String(`mock_op_${ops.length}_${account.id}`),
+    hash,
+    type: "DELEGATE",
+    value: BigNumber(0),
+    fee,
+    senders: [address],
+    recipients: [],
+    blockHash: genHex(64, rng),
+    blockHeight: account.blockHeight - Math.floor((Date.now() - date) / 900000),
+    accountId: account.id,
+    date,
+    extra
+  };
+
+  /** update account cosmosResources, spendableBalance and operations */
+  account.cosmosResources = cosmosResources;
```

Comment:
I prefer immutable way
```
return {
  ...account,
  cosmosResources,
  spendableBalance,
  ...
}
```

Reply:
weidly the operations aren't updated that way :shrug: